src/Week1/CreditCard.hs

```diff
@@ -0,0 +1,72 @@
+module Week1.CreditCard where
+
+
+{-
+ Exercise 1: We need to first find the digits of a number.
+
+ 1. toDigits should convert positive Integers to a list of digits. (For 0 or
+ negative inputs, toDigits should return the empty list.)
+
+ 2. toDigitsRev should do the same, but with the digits reversed.
+
+ Examples: toDigits 1234 == [1,2,3,4]
+           toDigitsRev 1234 == [4,3,2,1]
+           toDigits 0 == []
+           toDigits (-17) == []
+ -}
+toDigits :: Integer -> [Integer]
+toDigits n
+  | n < 1     = []
+  | otherwise = toDigits (n `div` 10) ++ [n `mod` 10]
+
+
+-- One implementation of toDigitsRev could just use the definition of toDigits
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev n = reverse $ toDigits n
+
+
+-- Or we can rewrite toDigitsRev which would follows a similar pattern to toDigits
+toDigitsRev' :: Integer -> [Integer]
+toDigitsRev' n
+  | n < 1     = []
+  | otherwise = n `mod` 10 : toDigitsRev' (n `div` 10)
+
+
+{-
+ Exercise 2: Once we have the digits in the proper order, we need to
+ double every other one.
+
+ Examples: doubleEveryOther [8,7,6,5] == [16,7,12,5]
+           doubleEveryOther [1,2,3] == [1,4,3]
+
+ Idea is to reverse the list first and then double every second element
+ from the beginning. Once doubled, reverse the list again.
+ -}
+doubleEverOther :: [Integer] -> [Integer]
+doubleEverOther = reverse . zipWith (*) (cycle [1, 2]) . reverse
```

Comment:
Why do we need to reverse the list?

Reply:
We actually need double every second element from the end!
So here's what I do:
- Reverse the list (so I can double every second from the beginning.)
- Iterate every two elements at once while multiplying them by 1 and 2 respectively.
- Reverse the list again. This essentially means that we have doubled every second element now from the end.

Reply:
I've added the implementation requirements in the comments @kwannoel

Reply:
Yes, I was initially confused too, but then I looked up the question and realised that the doubling takes place from the end of the list 😅 .

Also, this solution is correct, and usually the preferred approach, since using library functions is preferred to actually deconstructing the list and multiplying every other element by 2. But as food for thought, how would you implement the function using only pattern matching (and the reverse function)?

Reply:
@oishikg I've implemented this with pattern matching as well.

Reply:
@nitinprakash96 looks good to me!